File: LEETCODE/132_Pattern.java
class Solution {
    public boolean find132pattern(int[] nums) {
        for(int k =0;k<nums.length;k++){
            for(int j=0;j<k;j++){
                for(int i=0;i<j;i++){
                    if(nums[i]<nums[k] && nums[k]<nums[j]){
                        return true;
                    }
                }
            }
        }
        return false;
    }
}

File: leetcodes/132_Pattern.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
        int n = nums.size();
        int third = INT_MIN;
        vector<int> st;
        for (int i = n - 1; i >= 0; --i) {
            if (nums[i] < third) return true;
            while (!st.empty() && nums[i] > st.back()) {
                third = st.back();
                st.pop_back();
            }
            st.push_back(nums[i]);
        }
        return false;
    }
};